src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -91,14 +91,19 @@ private static IVecInt toVecInt(IntStream stream) {
 	}
 
 	// TODO: Find a way to sort versions of mods by suggestions and conflicts (not crucial, though)
-	public Map<String, ModCandidate> findCompatibleSet(Logger logger, Map<String, ModCandidateSet> modCandidateSetMap) throws ModResolutionException {
+	public Map<String, ModCandidate> findCompatibleSet(Logger logger, Map<String, ModCandidateSet> modCandidateSetMap, Set<Throwable> exceptions) {
 		// First, map all ModCandidateSets to Set<ModCandidate>s.
 		boolean isAdvanced = false;
 		Map<String, List<ModCandidate>> modCandidateMap = new HashMap<>();
 		Set<String> mandatoryMods = new HashSet<>();
 
 		for (ModCandidateSet mcs : modCandidateSetMap.values()) {
-			Collection<ModCandidate> s = mcs.toSortedSet();
+			Collection<ModCandidate> s = null;
+			try {
+				s = mcs.toSortedSet();
+			} catch (ModResolutionException e) {
+				exceptions.add(e);
```

Comment:
It might be a better idea to add a `continue` after this? Currently this will probably add the `null` collection to the `modCandidateMap`, which might blow up later on when it is iterated on.